src/main/kotlin/dev/sparky200/skriptnbt/skript/expressions/NbtOfHolderExpression.kt

```diff
@@ -0,0 +1,153 @@
+                if (theDelta == null) rhsError()
+                @Suppress("UNCHECKED_CAST")
+                if (expr.isSingle) set(event, expr.getSingle(event), theDelta as MutableCompound)
+                else expr.getArray(event).forEach { set(event, it, theDelta as MutableCompound) }
+            }
+            Changer.ChangeMode.ADD -> {
+                if (theDelta == null) rhsError()
+                @Suppress("UNCHECKED_CAST")
+                if (expr.isSingle) add(event, expr.getSingle(event), theDelta as MutableCompound)
+                else expr.getArray(event).forEach { add(event, it, theDelta as MutableCompound) }
+            }
+            Changer.ChangeMode.REMOVE -> {
+                if (theDelta == null) rhsError()
+                if (expr.isSingle) remove(event, expr.getSingle(event), theDelta)
+                else expr.getArray(event).forEach { remove(event, it, theDelta) }
+            }
+            else -> super.change(event, delta, mode)
+        }
+    }
+
+    private fun set(event: Event?, obj: Any?, delta: MutableCompound) = with (SkriptNbt.platform) {
+        when (obj) {
+            is Entity -> obj.setNbt(nbtTagOf(delta) as NbtCompound)
+            is String -> {
+                if (isFromFile) nbtToFile(Path(obj), nbtTagOf(delta) as NbtCompound)
+            }
+            else -> noNbtError(event)
+        }
+    }
+
+    private fun add(event: Event?, obj: Any?, delta: MutableCompound) = with (SkriptNbt.platform) {
+        when (obj) {
+            is Entity -> obj.setNbt(nbtTagOf(obj.getNbt().toMap() + delta.map) as NbtCompound)
+            // no-op (need to save the string NBT to variable first)
+            is String -> {}
+            else -> noNbtError(event)
+        }
+    }
+
+    private fun remove(event: Event?, obj: Any?, delta: Any) = with (SkriptNbt.platform) {
+        val theDelta: Collection<*> = when (delta) {
+            is MutableMap<*, *> -> delta.keys
+            is Collection<*> -> delta
+            is String -> listOf(delta)
+            else -> throw IllegalArgumentException("delta must be one of MutableMap, Collection, or String")
+        }
+        if (theDelta.isEmpty()) return@with
+        if (theDelta.first() !is String) throw IllegalArgumentException("delta must be or contain only Strings")
+
+        when (obj) {
+            is Entity -> obj.setNbt(NbtCompound(obj.getNbt().filterNot { it.key in theDelta }))
+            // no-op (need to save the string NBT to variable first)
+            is String -> {}
+            else -> noNbtError(event)
+        }
+    }
+
+    override fun init(
+        expressions: Array<out Expression<*>>,
+        matchedPattern: Int,
+        isDelayed: Kleenean,
+        parseResult: SkriptParser.ParseResult
+    ): Boolean {
+        if (!super.init(expressions, matchedPattern, isDelayed, parseResult)) return false
+        isFromFile = matchedPattern == 1
+        isFromText = matchedPattern == 2
+        return true
+    }
```

Comment:
convention is init -> convert -> acceptChange -> change -> getReturnType -> getPropertyName